### ISSUE TITLE
Prevents nixos-install from functioning

### ISSUE THREAD

**Issue by NovaViper** [viperML/nh#37]
I had this program included in my flake but when I try to install my system with `nixos-install --flake .#yoganva --root /mnt`, the stops with the error "/mnt/nix/store/realylongid-store/Cargo.toml: no such file or directory". only removing the package from my `home.packages` allowed for my flake to finish installing.

**Comment by viperML** [comment]
What was the nix version used to install the OS?

**Comment by NovaViper** [comment]
> What was the nix version used to install the OS?

Sorry for the late reply! The version I'm using is NixOS/nixpkgs@e35dcc04 in nixpkgs-unstable

**Comment by viperML** [comment]
So you built an ISO from that commit, fro. which you ran nixos-install ?

**Comment by viperML** [comment]
nixos-install is broken for flakes, but I believe the issue is fixed for installers with nix 2.17 or above

**Comment by NovaViper** [comment]
> So you built an ISO from that commit, from which you ran nixos-install ?

No, that's what my flake used, the iso I used is using the stable 23.05 (particularally, the iso named `nixos-plasma5-23.05.3376.4c8cf44c5b94-x86_64-linux`).

> nixos-install is broken for flakes, but I believe the issue is fixed for installers with nix 2.17 or above

Odd.. I've been able to use my flake to install with that command just fine. 🤷🏾‍♀️ After I removed nh from my list of packages to be installed, it allowed me to install NixOS using the flake.

**Comment by NovaViper** [comment]
Wait don't close it just yet, there's still the present issue of the Cargo.toml error that prevents the installer from running. It stems directly from nh and not the installer; why I was saying when I removed nh from the list of packages to install the installer continues.

**Comment by viperML** [comment]
This is an issue of `nixos-install` / `nix` than can't build nh properly

**Comment by NovaViper** [comment]
Ah ok. I'll pass over this issue over to the correct repos then. Thank you! 👍🏾

**Comment by iynaix** [comment]
Unfortunately, I'm still experiencing this issue on the 23.11 iso that ships with nix 2.18.1.

**Comment by viperML** [comment]
As I mentioned, I have no idea about the solution for this

**Comment by viperML** [comment]
I've changed the codepaths that import the Cargo files, and added a test to install NixOS. Let me know if it works properly now.

**Comment by iynaix** [comment]
Thanks, that fixed it! :tada:
